### ISSUE TITLE
LimitRange as part of Kubernetes core resources

### ISSUE THREAD

**Issue by Smorenoz** [backstage/backstage#11815]
## Feature Suggestions

Going through the `KubernetesFanOutHandler` service, I noticed that the `DEFAULT_OBJECTS` array (which pretty much are all the v1 core resources) is missing an entry for `limitranges` which makes no possible for the actual fetching of these resources. I also noticed that even though you're exporting the `KubernetesFanOutHandler` class on the file level, you're not including this on the export from the index, making not possible for a user to use this class to actually use the `builder pattern` you're introducing with `builder.setObjectsProvider`

## Possible Implementation

I believe you can address this including the `limitranges` on the `DEFAULT_OBJECTS` like:

```ts
export const DEFAULT_OBJECTS: ObjectToFetch[] = [
  {
    group: '',
    apiVersion: 'v1',
    plural: 'pods',
    objectType: 'pods',
  },
  {
    group: '',
    apiVersion: 'v1',
    plural: 'services',
    objectType: 'services',
  },
  {
    group: '',
    apiVersion: 'v1',
    plural: 'configmaps',
    objectType: 'configmaps',
  },
  {
    group: '', 
    apiVersion: 'v1',
    plural: 'limitranges',
    objectType: 'limitranges'
  },
  {
    group: 'apps',
    apiVersion: 'v1',
    plural: 'deployments',
    objectType: 'deployments',
  },
  {
    group: 'apps',
    apiVersion: 'v1',
    plural: 'replicasets',
    objectType: 'replicasets',
  },
  {
    group: 'autoscaling',
    apiVersion: 'v1',
    plural: 'horizontalpodautoscalers',
    objectType: 'horizontalpodautoscalers',
  },
  {
    group: 'batch',
    apiVersion: 'v1',
    plural: 'jobs',
    objectType: 'jobs',
  },
  {
    group: 'batch',
    apiVersion: 'v1',
    plural: 'cronjobs',
    objectType: 'cronjobs',
  },
  {
    group: 'networking.k8s.io',
    apiVersion: 'v1',
    plural: 'ingresses',
    objectType: 'ingresses',
  },
  {
    group: 'apps',
    apiVersion: 'v1',
    plural: 'statefulsets',
    objectType: 'statefulsets',
  },
  
];

```
And you will need to modify the `type KubernetesObjectTypes` like:
```ts
export type KubernetesObjectTypes =
  | 'pods'
  | 'services'
  | 'configmaps'
  | 'limitranges'
  | 'deployments'
  | 'replicasets'
  | 'horizontalpodautoscalers'
  | 'jobs'
  | 'cronjobs'
  | 'ingresses'
  | 'customresources'
  | 'statefulsets';

```

If you do not feel comfortable adding a new resource to this (maybe for performance issues) you can also just export all the classes needed to implement the desired `strategy` on the `builder pattern`, some of these classes as far as I dig into this are:

* KubernetesFanOutHandler
* KubernetesClientProvider
* KubernetesClinetBasedFetcher

And a final possible solution is just removing the validation on the `app.config.yaml` were using an empty group is not allowed anymore.

## Context

I'm currently working with Kubernetes backstage trying to present the resource limits for a pod, and I'm taking into count the scenarios where a `LimitRange` is present on a namespace, that's why I started to dig about how can I actually fetch this resources.

**Comment by freben** [comment]
@mclarke47 may know best but yeah it sounds to me like both adding this per default and making sure that the requisite types are exported, both make sense to contribute if you're up for it!

**Comment by Smorenoz** [comment]
Hey! I just created this PR that I hope will address this issue. https://github.com/backstage/backstage/pull/11938

**Comment by evankanderson** [comment]
/close

(It looks like this was fixed in #11938)
